SequencePhenotype.java

```diff
@@ -1,50 +1,95 @@
-import java.util.*;
+import java.lang.*;
+
+/**
+ * <b>SequencePhenotype</b> represents a phenotype.
+ * SequencePhenotypes are identified by their data contents.
+ * SequencePhenotype permits all data, except for <b>""</b> and <b>null</b>.
+ *
+ * @author Thien Tran
+ */
 public class SequencePhenotype implements Phenotype {
 
-    // constants
-    @SuppressWarnings("unchecked")
-    public final String[] NUCLEOBASES = new String[]{"A", "C", "G", "T"};
+    /**
+     * The valid letters that make up the sequence of this SequencePhenotype
+     */
+    public final char[] ALPHABET = "ABCDEFGHIJKLMNOPQRSTUVWXYZ".toCharArray();
```

Comment:
I thought we were moving to an amino-acid representation -- are we allowing all English alphabet now?